catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java

```diff
@@ -361,6 +363,15 @@ private void addObject(Map<String, Object> obj, String name, Object value) {
             metacard.put(CACHED, ISO_8601_DATE_FORMAT.print(new DateTime()));
         }
 
+        Attribute internalLocalResource = result.getMetacard()
```

Comment:
You could avoid any direct references to attribute names by filtering the attributes against the metacardtype. Anything not in the metacard type goes in this top-level section. That seems to be the pattern here.

Long term, we probably need to see about the idea of a set of attributes distinct from the metacardtype that are added during various processing stages.